src/state/swap/reducer.ts

```diff
@@ -33,10 +25,10 @@ export default createReducer<SwapState>(initialState, (builder) =>
       (state, { payload: { typedValue, recipient, field, inputCurrencyId, outputCurrencyId } }) => {
         return {
           [Field.INPUT]: {
-            currencyId: inputCurrencyId,
+            currencyId: inputCurrencyId || null,
```

Comment:
would `??` be better here in case `currencyId` is falsy? I'm assuming an id of `0` wouldn't be valid anyway but we probably shouldn't bake those assumptions into the code.

Reply:
dito in `swap/hooks` if so

Reply:
the initial check needs to set it to null if the currencyId is '', as it is parsed from the url query param. otherwise, sgtm.